ibis/backends/base/client.py

```diff
@@ -136,22 +139,5 @@ def table(self, name: str) -> ir.TableExpr:
         qualified_name = self._qualify(name)
         return self.client.table(qualified_name, self.name)
 
-    def list_tables(self, like: str = None) -> list:
-        """Return a list of all tables available for the current database.
-
-        Parameters
-        ----------
-        like : string, default None
-          e.g. 'foo*' to match all tables starting with 'foo'.
-
-        Returns
-        -------
-        list
-            A list with all tables available for the current database.
-        """
-        return self.client.list_tables(
-            like=self._qualify_like(like), database=self.name
-        )
-
-    def _qualify_like(self, like: str | None) -> str | None:
-        return like
+    def list_tables(self, like=None):
```

Comment:
why does this have a different signature?

Reply:
This is `Database.list_tables()`. I didn't add the `database` argument here, since it feels a bit strange to do `con.database('foo').list_tables(database='bar')`. I can add it if you want to have the same signature, but in any case, `Database` is set for deprecation, so no big deal.

Reply:
This operation is in a `Database` object.

What we have in this PR now is:
```python
>>> import ibis
>>> con = ibis.sqlite.connect(...)
>>> con.list_tables()  # `database` is the current database
>>> con.list_tables(database='bar')  # I can specify a different database when listing tables from the connection/client
>>> con.database('bar').list_tables()  # another way to list tables from a different database, equivalent to the previous line
>>> con.database('bar').list_tables(database='foobar')  # DOES NOT WORK. This could work if we make this signature the same as the rest.
```

What does need to be resolved? Do you want to make the last line work? I'm happy to add a `database` argument here if you want, but personally I don't think offering this to the user adds value.

Reply:
i actually don't care about the last line, instead about a consistent signature. I don't think the database arg adds *any* value. so either remove it everywhere or have it everywhere.

Reply:
Ok, I'll add it here for now, and will create an issue to discuss if there are better ways to support databases different than the default. Thanks!